app/src/models/app-menu.ts

```diff
@@ -189,7 +190,7 @@ function menuItemFromElectronMenuItem(menuItem: Electron.MenuItem): MenuItem {
   // Our menu items always have ids and Electron.MenuItem takes on whatever
   // properties was defined on the MenuItemOptions template used to create it
   // but doesn't surface those in the type declaration.
-  const id: string | undefined = (menuItem as any).id
+  const id: MenuIDs | undefined = (menuItem as any).id
```

Comment:
I don't believe all `menuItem.id` will actually be MenuIDs here. We use `MenuIDs` for all the menu items that we're going to be manipulating from the renderer (setting enabled/disabled, changing text, etc).

For all the other menu items we [generate an automatic unique identified](https://github.com/desktop/desktop/blob/452c89d0de300878e2320c5da7bdeb42e949d765/app/src/main-process/menu/ensure-item-ids.ts#L16) based on its position, label, or role.

So while the `any` cast here is valid for strings, it's not valid for `MenuIDs`

Reply:
ah thanks @niik, i totally missed the autogenerated ids logic! seems like this PR is not what we want then, so i'll close it.